### PR TITLE
Fix scaling problems in sixDofDragBehavior

### PR DESCRIPTION
**Pull request by carolhmj**
Related forum issue: https://forum.babylonjs.com/t/unselected-meshes-move-alongside-the-selected-mesh/46627/11?u=carolhmj

The resetting/setting of the parent in `_sceneRenderObserver` seems to be causing some instability in the scale decomposition for nodes with parents and certain rotation/scales, so instead of doing this process (which is also more costly in performance terms), the idea is to do all the operations in the node's local space.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14669/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14669/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14669/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14669/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14669/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14669/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14669/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14669/merge#BCU1XR#0

**Review by sebavan — COMMENTED** [review]
LGTM, I trust you on the maths :-) Let s merge after @Popov72 review

**Comment by Popov72** [comment]
Same for the maths :)

Maybe we can ping @CraigFeldspar as he is the original author of the behavior.

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Behaviors/Meshes/sixDofDragBehavior.ts [hunk 1]
```diff
@@ -81,18 +81,36 @@ export class SixDofDragBehavior extends BaseSixDofDragBehavior {
...
-                ownerNode.position.addInPlace(this._targetPosition.subtract(ownerNode.position).scale(this.dragDeltaRatio));
+                const deltaToAdd = TmpVectors.Vector3[0];
+                deltaToAdd.copyFrom(this._targetPosition).subtractInPlace(ownerNode.absolutePosition).scaleInPlace(this.dragDeltaRatio);
+                const deltaToAddTransformed = TmpVectors.Vector3[1];
+                deltaToAddTransformed.copyFrom(deltaToAdd);
+                // If the node has a parent, transform the delta to local space, so it can be added to the
+                // position in local space
+                if (ownerNode.parent) {
+                    const parentWorld = ownerNode.parent.getWorldMatrix();
+                    Vector3.TransformCoordinatesToRef(deltaToAdd, parentWorld, deltaToAddTransformed);
```
Comment (CraigFeldspar) [inline packages/dev/core/src/Behaviors/Meshes/sixDofDragBehavior.ts:92]: If it is a delta, shouldn't it be `TransformNormalToRef` ?

Comment (carolhmj) [reply packages/dev/core/src/Behaviors/Meshes/sixDofDragBehavior.ts:92]: You're right! Thanks for catching that.

**Pull request merged**